src/rgw/rgw_rest_s3.cc

```diff
@@ -5870,3 +5875,321 @@ bool rgw::auth::s3::S3AnonymousEngine::is_applicable(
 
   return route == AwsRoute::QUERY_STRING && version == AwsVersion::UNKNOWN;
 }
+
+
+using namespace s3selectEngine;
+const char* RGWSelectObj_ObjStore_S3::header_name_str[3] = {":event-type", ":content-type", ":message-type"};
+const char* RGWSelectObj_ObjStore_S3::header_value_str[3] = {"Records", "application/octet-stream", "event"};
+
+RGWSelectObj_ObjStore_S3::RGWSelectObj_ObjStore_S3():
+  s3select_syntax(std::unique_ptr<s3selectEngine::s3select>(new s3select())),
+  m_s3_csv_object(std::unique_ptr<s3selectEngine::csv_object>(((s3selectEngine::csv_object*)0))),
+  m_buff_header(std::unique_ptr<char>(new char[1000])),
```

Comment:
memory allocated with `new[]` has to be freed with `delete[]`, ie:
```c++
char* array = new char[1000];
delete[] array;
```

`std::unique_ptr` can do this for you if the template type is an array:
```c++
std::unique_ptr<char[]> m_buff_header;
m_buff_header = std::make_unique<char[]>(1000);
```